oci.go

```diff
@@ -22,13 +22,18 @@ import (
 const (
 	ociConfigFile     string      = "config.json"
 	ociConfigFileMode os.FileMode = 0444
+	ociConfigBasePath string      = "/run/libcontainer"
 )
 
-// writeSpecToFile writes the container's OCI spec to "dirname(spec.Root.Path)/config.json"
-// This effectively makes the parent directory a valid OCI bundle.
-func writeSpecToFile(spec *specs.Spec) error {
-	bundlePath := filepath.Dir(spec.Root.Path)
-	configPath := filepath.Join(bundlePath, ociConfigFile)
+// writeSpecToFile writes the container's OCI spec to "/run/libcontainer/<container-id>/config.json"
+// Note that the OCI bundle (rootfs) is at a different path
+func writeSpecToFile(spec *specs.Spec, containerId string) error {
+	configJsonDir := filepath.Join(ociConfigBasePath, containerId)
```

Comment:
Once the container terminated, was the file be removed?

Reply:
Hmm I'm not sure. However given that there could be `poststop` hooks which will run after the container has stopped, the config json might not be removed by the runtime.  @bergwolf would you have any idea ?

Reply:
I think we can do the cleanup in function: RemoveContainer  https://github.com/kata-containers/agent/blob/master/grpc.go#L1252

Reply:
I see that the `/run/` folder is a tmpfs mount. The config.json should get removed automatically along with other files that are there.

Reply:
Yes, that's only when the pod is stopped.  But if the pod is still live and users create/stop many containers, then there would be much unused spec files left there.

Reply:
I understand the scenario now. Thanks @lifupan. So in this case should we also handle cleanup of state.json that is kept under `/run/libcontainer/<id>`? And if yes, does it make sense to handle cleanup in a separate patch or club with this one ?
Also the impact of removing config.json on container `poststop` hook needs to be checked. Let me know your thoughts @lifupan

Reply:
Could you check if `libcontainer` removes `/run/libocntainer/<id>` entirely upon container removal? If so, there is nothing to be done in the agent code.

Reply:
@bpradipt Lets make sure we clean up correctly in this patch.

Reply:
Hi @bpradipt  @bergwolf  The dir  /run/libocntainer/<id>  was created by this patch in agent, I don't think libcontainer would do cleanup of this dir.  BTW, the libcontainer would run he poststophooks in container's Destroy function at https://github.com/kata-containers/agent/blob/master/grpc.go#L1264 or https://github.com/kata-containers/agent/blob/master/grpc.go#L1279, so you can just do the cleanup behind that destroy.